File: leetcode_ts/src/190.颠倒二进制位.ts
/*
 * @lc app=leetcode.cn id=190 lang=typescript
 *
 * [190] 颠倒二进制位
 */

// @lc code=start
function reverseBits(n: number): number {
  let str = n.toString(2);
  let arr = new Array(32).fill(0);
  for(let i = str.length;i >= 0;i--){
    let j = str.length - 1 - i;
    arr[j] = str[i];
  }
  str = arr.join("");
  let res = parseInt(str,2);
  return res;
};
// @lc code=end

reverseBits(parseInt("00000010100101000001111010011100",2));

File: leetcode_go/pkg/algorithm/190.颠倒二进制位.go
package algorithm

import (
	"strconv"
)

/*
 * @lc app=leetcode.cn id=190 lang=golang
 *
 * [190] 颠倒二进制位
 */

// @lc code=start
func reverseBits(num uint32) uint32 {
	str := strconv.FormatInt(int64(num), 2)
	arr := [32]string{}
	for i := len(str) - 1; i >= 0; i-- {
		j := len(str) - 1 - i
		arr[j] = string(str[i])
	}
	str = ""
	for i := 0; i < 32; i++ {
		str += arr[i]
	}
	offset := 32 - len(str)
	for i := 0; i < offset; i++ {
		str += "0"
	}
	res, _ := strconv.ParseInt(str, 2, 64)
	return uint32(res)
}

// @lc code=end

func ReverseBits(num uint32) uint32 {
	return reverseBits(num)
}
